File: Server/Utils/Destination.js
class Destination {
    constructor() {
        this.toPlanet = false;
        this.toPlayer = false;
        this.planetName = null;
        this.playerName = null;
    }
    static Deserialize(jsonString) {
        var newDestination = new Destination();
        var parsedJson = JSON.parse(jsonString);
        Object.keys(parsedJson).forEach(key => {
            newDestination[key] = parsedJson[key]
        });
        return newDestination;
    }
}
module.exports = Destination

File: Space Traveling/Assets/Scripts/Destination.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
class Destination
{
    public bool toPlanet = false;
    public bool toPlayer = false;
    public string planetName;
    public string playerName;

    public Destination()
    {

    }
    public string Serialize()
    {

        return JsonConvert.SerializeObject(this);
        //return JsonConvert.SerializeObject(this);

    }
    public static Destination Deserialize(string jsonString)
    {
        JObject dict = JObject.Parse(jsonString);
        Destination destination = JsonConvert.DeserializeObject<Destination>(jsonString);
        return destination;
    }
}